qlib/data/dataset/handler.py

```diff
@@ -243,8 +243,9 @@ def _fetch_data(
                 data_df = fetch_df_by_col(data_df, col_set)
             else:
                 # Fetch column  first will be more friendly to SepDataFrame
-                data_df = fetch_df_by_col(data_df, col_set)
+                # Fetch index first will be more friendly to Memory. Because pandas will allocate new memory for the intermediate dataframe, and the extra dataframe will be smaller.
```

Comment:
We have a  class like  this
https://github.com/microsoft/qlib/blob/main/qlib/contrib/data/utils/sepdf.py#L17

For this class, it is more friendly to fetch the column first.

So it is hard to decide which is better.
I suggest the following solutions (any one of them looks reasonable to me)
-  add a special method like the `default_fetch_order` method for  SepDataFrame and check 
-  Add `default_fetch_order` parameter for the handler and keep the default behavior the same as before.